### ISSUE TITLE
add gtk theme and icon setting

### ISSUE THREAD

**Issue by librewish** [WayfireWM/wcm#10]
it will be very convenient for having 
gtk theming gtk icon and gtk cursor theming option in wcm.
which will do
for example

gsettings set org.gnome.desktop.interface gtk-theme Matcha-dark-sea

gsettings set org.gnome.desktop.interface icon-theme Papirus-Dark

gsettings set org.gnome.desktop.interface cursor-theme xcursor-breeze

**Comment by soreau** [comment]
wcm is for configuring wayfire which means it's centered around modifying wayfire.ini config file. It would require special casing to support this. This doesn't mean it's out of the question to implement though.

Is there a way to get a list of all available supported themes for any of these categories?

**Comment by librewish** [comment]
the themes are located in
 /usr/share/themes

and in ~/.local/share/themes
for icons
/usr/share/icons
and
~/.local/share/icons
the folder name is the theme name.

**Comment by soreau** [comment]
This doesn't give me much incentive to support this.

**Comment by librewish** [comment]
but gnome tweaks require gnome shell to be running

**Comment by ammen99** [comment]
@librewish This is not true. I have used gnome-tweaks in Wayfire.

**Comment by librewish** [comment]
@ammen99 do you have gnome-shell installed because i dont have it
and gnome-tweaks wont run without it.
i dont have gnome-control-center installed either

**Comment by ammen99** [comment]
Yes, I have GNOME installed (I use it rarely to check whether applications behave differently than in wayfire)

**Comment by lukefromdc** [comment]
In mate-control-center's mate-appearance-properties from GIT master(1.29 as of now), we have code to write to the GNOME gsettings values as well as the MATE gsettings values for GTK theme, icon theme, font etc when a wayland session is detected. This is to support the wayfire-based MATE wayland session. Since these are just gsettings values followed by GTK, there are no issues with allowing multiple programs to be used to write to them. We have dconf-editior, mate-control-center, and gnome-tweaks all able to coexist and write these values, which are then immediately applied.

I don't see any reason it would be a problem to add a similar setting to WCM although the complex theme listing and thumbnailing code we use in MATE need not be reimplemented. A simple combobox listing the themes by name would be enough.

It took me a while when I first started playing with wayfire as a compositor for MATE to even figure out that GTK under wayland follows the GNOME values for these, presumably many don't know that. Although this is hardcoded and compositors from my understanding can't change that, their configuration utilities (or anyone else's) can write to the gsettings values.

**Comment by soreau** [comment]
I probably should mention that there is a [gsettings backend](https://codeberg.org/valpackett/wf-gsettings) for wayfire. In fact, any [settings backend](https://github.com/WayfireWM/wayfire/blob/master/src/main.cpp#L33) can be written for wayfire. You can back it with something other than the [default wayfire.ini backend](https://github.com/WayfireWM/wayfire/blob/master/src/default-config-backend.cpp).
